components/scream/src/diagnostics/tests/relative_humidity_tests.cpp

```diff
@@ -0,0 +1,188 @@
+  ekat::ParameterList params;
+  params.set<std::string>("Diagnostic Name", "Relative Humidity");
+  params.set<std::string>("Grid", "Point Grid");
+  register_diagnostics();
+  auto& diag_factory = AtmosphereDiagnosticFactory::instance();
+  auto diag = diag_factory.create("RelativeHumidity",comm,params);
+  diag->set_grids(gm);
+
+
+  // Set the required fields for the diagnostic.
+  std::map<std::string,Field> input_fields;
+  for (const auto& req : diag->get_required_field_requests()) {
+    Field f(req.fid);
+    auto & f_ap = f.get_header().get_alloc_properties();
+    f_ap.request_allocation(packsize);
+    f.allocate_view();
+    const auto name = f.name();
+    f.get_header().get_tracking().update_time_stamp(t0);
+    diag->set_required_field(f.get_const());
+    REQUIRE_THROWS(diag->set_computed_field(f));
+    input_fields.emplace(name,f);
+  }
+
+  // Initialize the diagnostic
+  diag->initialize(t0,RunType::Initial);
+
+  // Run tests
+  {
+    // Construct random data to use for test
+    // Get views of input data and set to random values
+    const auto& T_mid_f = input_fields["T_mid"];
+    const auto& T_mid_v = T_mid_f.get_view<Pack**>();
+    const auto& p_mid_f = input_fields["p_mid"];
+    const auto& p_mid_v = p_mid_f.get_view<Pack**>();
+    const auto& qv_f = input_fields["qv"];
+    const auto& qv_v = qv_f.get_view<Pack**>();
+
+    for (int icol=0;icol<ncols;icol++) {
+      const auto& T_sub = ekat::subview(T_mid_v,icol);
+      const auto& p_sub = ekat::subview(p_mid_v,icol);
+      const auto& qv_sub = ekat::subview(qv_v,icol);
+      ekat::genRandArray(dview_as_real(temperature), engine, pdf_temp);
+      ekat::genRandArray(dview_as_real(pressure),    engine, pdf_pres);
+      ekat::genRandArray(dview_as_real(qv),    engine, pdf_qv);
+      Kokkos::deep_copy(T_sub,temperature);
+      Kokkos::deep_copy(p_sub,pressure);
+      Kokkos::deep_copy(qv_sub,qv);
+    }
+
+    // Run diagnostic and compare with manual calculation
+    diag->compute_diagnostic();
+    const auto& diag_out = diag->get_diagnostic();
+    Field rh_f = diag_out.clone();
+    rh_f.deep_copy<double,Host>(0.0);
+    rh_f.sync_to_dev();
+    const auto& rh_v = rh_f.get_view<Pack**>();
+    using physics = scream::physics::Functions<Real, DefaultDevice>;
+    using Smask = ekat::Mask<Pack::n>;
+    Smask range_mask(true);
+    Kokkos::parallel_for("", policy, KOKKOS_LAMBDA(const MemberType& team) {
+      const int icol = team.league_rank();
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team,num_mid_packs), [&] (const Int& jpack) {
+      auto qv_sat_l = physics::qv_sat(T_mid_v(icol,jpack), p_mid_v(icol,jpack), false, range_mask);
+        rh_v(icol,jpack) = qv_v(icol,jpack)/qv_sat_l;
+//        rh_v(icol,jpack) = PF::calculate_theta_from_T(T_mid_v(icol,jpack),p_mid_v(icol,jpack));
+      });
+      team.team_barrier();
+    });
+    Kokkos::fence();
+    REQUIRE(views_are_equal(diag_out,rh_f));
+  }
+ 
+  // Finalize the diagnostic
```

Comment:
Similar to LWCF, are there any physical or mathematical tests we can add for RH?